src/vanilla/shallow.ts

```diff
@@ -39,7 +39,7 @@ export function shallow<T>(objA: T, objB: T) {
   }
   for (let i = 0; i < keysA.length; i++) {
     if (
-      !Object.prototype.hasOwnProperty.call(objB, keysA[i] as string) ||
+      !Object.hasOwn(objB, keysA[i] as string) ||
```

Comment:
I generally like the new syntax, but technically, we still support ie11.
https://github.com/pmndrs/zustand/blob/3c6dc2d12b0d2a2d5f403fe6ddad1053055ab72b/rollup.config.js#L107

So, let's make this change to v5 instead of v4. ie11 will break but it's probably okay.

Reply:
I'm down with that this should be targeted to v5

Reply:
@dai-shi @onlined in any case we need to keep in mind this -> https://caniuse.com/?search=hasOwn

Reply:
Updated base branch.